src/renderers/dom/client/ReactEventListener.js

```diff
@@ -60,29 +62,59 @@ PooledClass.addPoolingTo(
 );
 
 function handleTopLevelImpl(bookKeeping) {
-  var topLevelTarget = ReactMount.getFirstReactDOM(
-    getEventTarget(bookKeeping.nativeEvent)
-  ) || window;
-
-  // Loop through the hierarchy, in case there's any nested components.
-  // It's important that we build the array of ancestors before calling any
-  // event handlers, because event handlers can modify the DOM, leading to
-  // inconsistencies with ReactMount's node cache. See #1105.
-  var ancestor = topLevelTarget;
-  while (ancestor) {
-    bookKeeping.ancestors.push(ancestor);
-    ancestor = findParent(ancestor);
-  }
+  var path = bookKeeping.nativeEvent.path;
+  var topLevelTargetID;
+  if (path != null) {
+    var oldRootId = undefined;
+    var currentNativeTarget = undefined;
```

Comment:
Can `currentNativeTarget` ever be undefined in real cases? It seems like it should always be defined from a type system perspective and therefore we can ensure that we write the code in a way that it is. E.g. by initializing it to `var currentNativeTarget = path[0];` and then `currentNativeTarget = path[j +1]` once we hit a document fragment. That way we ensure that we don't make a mistake in our type logic.